open_lm/model.py

```diff
@@ -305,20 +309,42 @@ def create_params(args):
     # These args are managed separately by the argparser
     # If a parameter is in the model config, regardless of the args, we use the config parameters
     # If a parameter is not in the model config, we use the args parameter
-    return Params(
-        dim=cfg["hidden_dim"],
-        n_layers=cfg["n_layers"],
-        n_heads=cfg["n_heads"],
-        seq_len=cfg["seq_len"],
-        vocab_size=cfg["vocab_size"],
-        post_embed_norm=cfg["post_embed_norm"],
-        weight_tying=cfg["weight_tying"],
-        norm_type=get_norm_class(cfg.get("model_norm", args.model_norm)),
-        apply_qk_norm=cfg.get("qk_norm", args.qk_norm),
-        positional_embedding_type=cfg.get("positional_embedding_type", args.positional_embedding_type),
-        ffn_type=cfg.get("ffn_type", args.ffn_type),
-    )
+
+    if 'mamba' in args.model:
+        return {
+            "d_model": cfg['d_model'],
+            "n_layer": cfg['n_layer'],
+            "vocab_size": cfg['vocab_size'],
+            "seq_len": cfg['seq_len'],
+        }
+    else:
+        return Params(
+            dim=cfg["hidden_dim"],
+            n_layers=cfg["n_layers"],
+            n_heads=cfg["n_heads"],
+            seq_len=cfg["seq_len"],
+            vocab_size=cfg["vocab_size"],
+            post_embed_norm=cfg["post_embed_norm"],
+            weight_tying=cfg["weight_tying"],
+            norm_type=get_norm_class(cfg.get("model_norm", args.model_norm)),
+            apply_qk_norm=cfg.get("qk_norm", args.qk_norm),
+            positional_embedding_type=cfg.get("positional_embedding_type", args.positional_embedding_type),
+            ffn_type=cfg.get("ffn_type", args.ffn_type),
+        )
 
 
 def create_model(args):
-    return Transformer(create_params(args))
+    if 'mamba' in args.model:
```

Comment:
This is fine for now, but would be nice if we could minimize how many places we have if/else. Maybe create_params(args) and then have a Mamba subclass that takes that in and handles all the modifications (vocab_size, popping seqlen, etc.)